### PR TITLE
[nova] Fixes for NOVA TV

### LINKED ISSUE #23672: Download from novaplus.nova.cz does not work anymore
## Checklist

- [x] I'm reporting a broken site support
- [x] I've verified that I'm running youtube-dl version **2020.01.01**
- [x] I've checked that all provided URLs are alive and playable in a browser
- [x] I've checked that all URLs and arguments with special characters are properly quoted or escaped
- [x] I've searched the bugtracker for similar issues including closed ones

## Verbose log

```
[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'https://novaplus.nova.cz/porad/ulice/epizoda/41089-ulice-3802-dil', u'--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.01.01
[debug] Python version 2.7.13 (CPython) - Darwin-19.2.0-x86_64-i386-64bit
[debug] exe versions: none
[debug] Proxy map: {}
[Nova] 41089-ulice-3802-dil: Downloading webpage
[NovaEmbed] I4FOkPvRUA6: Downloading webpage
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "./youtube-dl-2.dms/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "./youtube-dl-2.dms/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "./youtube-dl-2.dms/youtube_dl/extractor/nova.py", line 66, in _real_extract
    self._sort_formats(formats)
  File "./youtube-dl-2.dms/youtube_dl/extractor/common.py", line 1327, in _sort_formats
    raise ExtractorError('No video formats found')
ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.[debug] System config: []
[debug] User config: []
[debug] Custom config: []
[debug] Command-line args: [u'https://novaplus.nova.cz/porad/ulice/epizoda/41089-ulice-3802-dil', u'--verbose']
[debug] Encodings: locale UTF-8, fs utf-8, out UTF-8, pref UTF-8
[debug] youtube-dl version 2020.01.01
[debug] Python version 2.7.13 (CPython) - Darwin-19.2.0-x86_64-i386-64bit
[debug] exe versions: none
[debug] Proxy map: {}
[Nova] 41089-ulice-3802-dil: Downloading webpage
[NovaEmbed] I4FOkPvRUA6: Downloading webpage
ERROR: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
Traceback (most recent call last):
  File "./youtube-dl-2.dms/youtube_dl/YoutubeDL.py", line 796, in extract_info
    ie_result = ie.extract(url)
  File "./youtube-dl-2.dms/youtube_dl/extractor/common.py", line 530, in extract
    ie_result = self._real_extract(url)
  File "./youtube-dl-2.dms/youtube_dl/extractor/nova.py", line 66, in _real_extract
    self._sort_formats(formats)
  File "./youtube-dl-2.dms/youtube_dl/extractor/common.py", line 1327, in _sort_formats
    raise ExtractorError('No video formats found')
ExtractorError: No video formats found; please report this issue on https://yt-dl.org/bug . Make sure you are using the latest version; type  youtube-dl -U  to update. Be sure to call youtube-dl with the --verbose flag and include its complete output.
```

## Description

The download of video from novaplus.nova.cz does not work anymore. The URL https://novaplus.nova.cz/porad/ulice/epizoda/41089-ulice-3802-dil is playable in browser but not downloable. It worked until end of last year, probably the site was changed.

### PR DESCRIPTION
**Pull request by jtrmal**
## Please follow the guide below

- You will be asked some questions, please read them **carefully** and answer honestly
- Put an `x` into all the boxes [ ] relevant to your *pull request* (like that [x])
- Use *Preview* tab to see how your *pull request* will actually look like

---

### Before submitting a *pull request* make sure you have:
- [x] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [x] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [x] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [x] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [x] Bug fix
- [ ] Improvement
- [ ] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Explanation of your *pull request* in arbitrary form goes here. Please make sure the description explains the purpose and effect of your *pull request* and is worded well enough to be understood. Provide as much context and examples as possible.

This should resolve #23672
Some of the other tests seem to fail, (but they were failing originally) and I don't understand the pages for which it should work enough and the code enough (and the youtube-dl conventions enough) to fix those.

There are two comments/questions in the code -- I;d appreciate the input

### CODE REVIEW

File: youtube_dl/extractor/nova.py [hunk 1]
```diff
@@ -152,14 +171,30 @@ def _real_extract(self, url):
...
-            return self.url_result(
+            info = {
+                'description': description,
+                'upload_date': upload_date
+            }
+            info.update(self.url_result(
                 'https://media.cms.nova.cz/embed/%s' % embed_id,
-                ie=NovaEmbedIE.ie_key(), video_id=embed_id)
+                ie=NovaEmbedIE.ie_key(), video_id=embed_id))
+            return info
```
Comment (jtrmal) [inline youtube_dl/extractor/nova.py]: My main issue is that I don't know how to make the 'url result' type to inherit the description and upload type from  the "upper" container. The test will fail if I uncomment 'description' and 'upload_date' in the TEST dictionary...
But when I run the youtube-dl  with my modifications, all seem to work ok.

File: youtube_dl/extractor/nova.py [hunk 2]
```diff
@@ -100,6 +104,21 @@ class NovaIE(InfoExtractor):
...
+            'title': '2020-01-08 Televizní noviny',
+            # 'description': 're:.*Sportovní noviny, Počasí Pořad je opatřen audiodeskripcí.*',
+            'thumbnail': r're:https?://.*\.jpg(\?.*)?',
+            # 'upload_date': '20200108'
+        },
+        'params': {
+            # rtmp download
+            'skip_download': True,
+        },
+        'add_ie': [NovaEmbedIE.ie_key()],
```
Comment (jtrmal) [inline youtube_dl/extractor/nova.py]: I'm not sure if I need this

File: youtube_dl/extractor/nova.py (hunk 1 above)
Comment (dstftw) [reply youtube_dl/extractor/nova.py]: You should use info dict with `'_type': 'url_transparent'` instead.

File: youtube_dl/extractor/nova.py (hunk 2 above)
Comment (dstftw) [reply youtube_dl/extractor/nova.py]: Only if extraction goes through delegation to `NovaEmbedIE`.

File: youtube_dl/extractor/nova.py [hunk 3]
```diff
@@ -43,6 +43,10 @@ def _real_extract(self, url):
 
         formats = []
         for format_id, format_list in bitrates.items():
+            if format_id == 'hls':
+                formats.extend(self._extract_m3u8_formats(
+                    format_list, video_id, ext='mp4', m3u8_id='hls', fatal=False))
```
Comment (dstftw) [inline youtube_dl/extractor/nova.py]: You must validate `format_list`. Read coding conventions.

File: youtube_dl/extractor/nova.py [hunk 4]
```diff
@@ -152,14 +171,30 @@ def _real_extract(self, url):
...
+        elif site == 'fanda':
+            upload_date = unified_strdate(self._search_regex(
+                r'<span class="date_time">(\d{1,2}\.\d{1,2}\.\d{4})', webpage, 'upload date', default=None))
+        else:
+            upload_date = None
+
         # novaplus
         embed_id = self._search_regex(
-            r'<iframe[^>]+\bsrc=["\'](?:https?:)?//media\.cms\.nova\.cz/embed/([^/?#&]+)',
+            r'<iframe[^>]+\bsrc=[\"\'](?:https?:)?//media\.cms\.nova\.cz/embed/([^/?#&]+)',
```
Comment (dstftw) [inline youtube_dl/extractor/nova.py]: Revert all useless changes.

File: youtube_dl/extractor/nova.py (hunk 2 above)
Comment (jtrmal) [reply youtube_dl/extractor/nova.py]: thanks

File: youtube_dl/extractor/nova.py (hunk 1 above)
Comment (jtrmal) [reply youtube_dl/extractor/nova.py]: addressed -- now passes the test with flying colors.

File: youtube_dl/extractor/nova.py (hunk 3 above)
Comment (jtrmal) [reply youtube_dl/extractor/nova.py]: This I didn't figure out -- I don't quite understand what you mean, could you please be more specific?

**Comment by jtrmal** [comment]
@dstftw thanks for the feedback, I didn't figure out what do you mean by validation of the format_list, could you be more specific? Sorry...
The other stuff is done.

File: youtube_dl/extractor/nova.py (hunk 3 above)
Comment (dstftw) [reply youtube_dl/extractor/nova.py]: `format_list` is not guaranteed to be an URL.

Comment (jtrmal) [reply youtube_dl/extractor/nova.py]: Thanks and apologies, resolved.

**Pull request closed without merging**